### DISCUSSION TITLE
How to set up boundaryObject for a bipedal robot construction?

### DISCUSSION THREAD

**Discussion by huiduan-25** [cyberbotics/webots#5250] [Q&A]
Hello, 
I build a bipedal robot in SolidWorks and import the model into Webots. All the parts are lost their mates and become transform nodes. I change them to solid nodes and move them into new solid nodes shown in Figure1. These new solid nodes are in a robot node. I also create a transform node called body_trans and a Box shape node in the children. So, the boundaryObject can select the body_trans node or any transform node shown in Figure 2. 

I am not sure how I can make the boundaryObject select the robot node, not a simple box or a specific transform node of part.

Thank you very much for your kind help!
<img width="638" alt="image" src="https://user-images.githubusercontent.com/69326215/191341716-756bb149-f742-47f1-b326-fa6a95058788.png">

**Comment by omichel** [comment]
You should define a bounding object for each articulated component of the robot. I would recommend you follow the [Webots tutorials](https://cyberbotics.com/doc/guide/tutorials) to learn how to create bounding objects for a robot, especially tutorials 5 and 6.

**Reply by huiduan-25** [reply]
Hi, Michel

Thank you for your reply. Most of components of the robot have complex shape, I am not sure how to define their boundaryObject by adding shape node. Could you please give me more information?

Thank you

**Reply by omichel** [reply]
You should create a compound bounding object as explained in [tutorial 5](https://cyberbotics.com/doc/guide/tutorial-5-compound-solid-and-physics-attributes). Such a bounding object combines several primitives (spheres, cylinders, boxes, capsules, etc.) into a single bounding object.
